### PR TITLE
feat: enable GitHub actions

### PR DESCRIPTION
**Pull request by sendilkumarn**
Closes #811 

This converts most of the Travis build to GitHub Actions. 

cc: @motlin @nikhilnanivadekar @donraab

### CODE REVIEW

**Comment by nikhilnanivadekar** [comment]
I agree we shouldnt cache anything, especially Eclipse Collections jars.

File: .github/workflows/acceptance.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      - name: Set Maven Wrapper
+        run: mvn -N io.takari:maven:0.7.7:wrapper -Dmaven=3.6.1
+      - name: Set JDK ${{ matrix.java-version }}
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java-version }}
+      - name: Enable Maven Cache
+        uses: actions/cache@v1
+        with:
+          path: ~/.m2/repository
```
[the remark this answers is not in the document]
Comment (sendilkumarn) [reply .github/workflows/acceptance.yml]: Updated this.

**Comment by sendilkumarn** [comment]
@donraab Updated the workflow name and added the necessary badges 👍

**Comment by motlin** [comment]
I said we should avoid caching `~/.m2/repository/org/eclipse/collections` and @nikhilnanivadekar said we shouldn't cache anything, and @sendilkumarn it looks like you made that change. I do believe we should cache everything other than ~/.m2/repository/org/eclipse/collections, but it was really a question. Why would we choose to cache nothing? Ease of implementation?

**Comment by sendilkumarn** [comment]
If we find that our builds are slower, then we will think about caching. But for now the builds are faster even without caching anything explicitly.

**Pull request merged**